File: app/src/main/java/com/example/cobe/cryptonews/presentation/MainInterface.java
package com.example.cobe.cryptonews.presentation;


import com.example.cobe.cryptonews.model.Article;

import java.util.List;

/**
 * Created by cobe on 06/04/2018.
 */

public interface MainInterface {

    interface View {

        void setSearchError();

        void startSearchActivity(String text);

        void startSearchActivity(String text, String date);

        void setDateError();

        void setArticlesFailure();

        void showArticles(List<Article> articles);

        void startArticleDetails(String url);
    }

    interface Presenter extends BasePresenter<View> {

        void onArticleInputSearch(String text);

        void onArticleDateSearch(String text, String date);

        void getArticles();

        void articleDetails(String url);
    }
}

File: app/src/main/java/com/example/cobe/cryptonews/presentation/MainInterface.kt
package com.example.cobe.cryptonews.presentation

import com.example.cobe.cryptonews.model.Article

/**
 * Created by cobe on 11/04/2018.
 */
interface MainInterface {
    interface View {
        fun setSearchError()

        fun startSearchActivity(text: String)

        fun startSearchActivity(text: String, date: String)

        fun setDateError()

        fun setArticlesFailure()

        fun showArticles(articles: List<Article>)

        fun startArticleDetails(url: String)
    }

    interface Presenter : BasePresenter<View> {

        fun onArticleInputSearch(text: String)

        fun onArticleDateSearch(text: String, date: String)

        fun getArticles()

        fun articleDetails(url: String)
    }
}